gtk4/src/expression.rs

```diff
@@ -0,0 +1,451 @@
+            ))
+        }
+    }
+}
+
+define_expression!(
+    ObjectExpression,
+    ffi::GtkObjectExpression,
+    ffi::gtk_object_expression_get_type
+);
+
+impl ObjectExpression {
+    #[doc(alias = "gtk_property_expression_new")]
+    pub fn new<T: IsA<Object>>(object: &T) -> Self {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gtk_object_expression_new(
+                object.as_ref().to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gtk_object_expression_get_object")]
+    pub fn get_object(&self) -> Option<Object> {
+        skip_assert_initialized!();
+        unsafe { from_glib_none(ffi::gtk_object_expression_get_object(self.to_glib_none().0)) }
+    }
+}
+
+define_expression!(
+    ClosureExpression,
+    ffi::GtkClosureExpression,
+    ffi::gtk_closure_expression_get_type
+);
+
+impl ClosureExpression {
+    #[doc(alias = "gtk_closure_expression_new")]
+    pub fn new(value_type: Type, closure: glib::Closure, params: &[Expression]) -> Self {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gtk_closure_expression_new(
+                value_type.to_glib(),
+                closure.to_glib_none().0,
+                params.len() as u32,
+                params.to_glib_full(),
+            ))
+        }
+    }
+}
+
+define_expression!(
+    ConstantExpression,
+    ffi::GtkConstantExpression,
+    ffi::gtk_constant_expression_get_type
+);
+
+impl ConstantExpression {
+    #[doc(alias = "gtk_constant_expression_new")]
+    pub fn new<V: ToValue>(value: &V) -> Self {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gtk_constant_expression_new_for_value(
+                value.to_value().to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gtk_constant_expression_new_for_value")]
+    pub fn for_value(value: &Value) -> Self {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gtk_constant_expression_new_for_value(
+                value.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gtk_constant_expression_get_value")]
+    pub fn get_value(&self) -> Value {
+        unsafe {
+            from_glib_none(ffi::gtk_constant_expression_get_value(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use glib::StaticType;
+
+    use super::*;
+
+    #[test]
+    fn test_pspec_expression() {
+        crate::init().unwrap();
+
+        let _pspec = glib::ParamSpec::expression(
+            "expression",
+            "Expression",
+            "Some Expression",
+            glib::ParamFlags::CONSTRUCT_ONLY | glib::ParamFlags::READABLE,
+        );
+    }
+
+    #[test]
+    fn test_property_expression() {
+        crate::init().unwrap();
+        let expr =
+            PropertyExpression::new(crate::FontChooser::static_type(), NONE_EXPRESSION, "font");
```

Comment:
It doesn't like the `NONE_EXPRESSION` here at all because the former expect `AsRef<Expression>` and the later `Option<&Expression>` which doesn't work because `Expression` doesn't implement `AsRef<Expression>`.

Reply:
Well, you don't implement `AsRef<Expression>` for any of the types :) Why not? You only implement `Deref`, so maybe just add `AsRef` in addition

Reply:
My bad, I somehow thought I already did.